build/openwisp_base/Dockerfile

```diff
@@ -13,33 +15,32 @@ FROM SYSTEM AS PYTHON
 # 1. Twisted: gcc
 # 2. Postgresql: libpq-dev
 RUN apt update && \
-    apt install --yes --no-install-recommends git gcc libpq-dev && \
+    apt install --yes --no-install-recommends gcc libpq-dev && \
     rm -rf /var/lib/apt/lists/* /root/.cache/pip/* /tmp/*
 
 ENV PYTHONPATH=/install/lib/python3.7/site-packages
 # Python Packages
-RUN pip install --editable git+git://github.com/openwisp/openwisp-controller#egg=openwisp_controller \
+RUN pip install --upgrade https://github.com/openwisp/openwisp-controller/tarball/master \
                 --prefix='/install'
-RUN pip install --editable git+git://github.com/openwisp/openwisp-radius#egg=openwisp_radius \
-                --prefix='/install'
-RUN pip install --editable git+git://github.com/openwisp/openwisp-network-topology#egg=openwisp_network_topology \
+RUN pip install --upgrade https://github.com/openwisp/openwisp-network-topology/tarball/master \
                 --prefix='/install'
 RUN pip install --upgrade https://github.com/openwisp/django-freeradius/tarball/master \
                 --prefix='/install'
+RUN pip install --upgrade https://github.com/openwisp/openwisp-radius/tarball/master \
+                --prefix='/install'
 RUN pip install --upgrade https://github.com/openwisp/openwisp-users/tarball/master \
                 --prefix='/install'
 RUN pip install --upgrade https://github.com/openwisp/django-netjsonconfig/tarball/master \
                 --prefix='/install'
 RUN pip install --upgrade https://github.com/openwisp/django-x509/tarball/master \
                 --prefix='/install'
-RUN pip install --editable git+git://github.com/openwisp/openwisp-utils#egg=openwisp_utils \
+RUN pip install --upgrade https://github.com/openwisp/openwisp-utils/tarball/master \
                 --prefix='/install'
 RUN pip install --prefix='/install' asgi_redis service_identity django-redis psycopg2 uwsgi \
-                                    sentry-sdk supervisor celery django-celery-beat redis
+                                    sentry-sdk supervisor celery django-celery-beat redis==2.10.6
```

Comment:
For context: redis is set to 2.10.6 because v3 is not compatible with django-channels.